internal/metrics/metrics.go

```diff
@@ -3,8 +3,6 @@ package metrics
 import (
 	"time"
 
-	"github.com/RHEnVision/provisioning-backend/internal/kafka"
-	"github.com/RHEnVision/provisioning-backend/internal/models"
 	"github.com/RHEnVision/provisioning-backend/internal/version"
 	"github.com/prometheus/client_golang/prometheus"
```

Comment:
This was the culprit of import cycle problems. The metrics package will be in wide use througout the whole app and it is important to keep the imports at minimum. Therefore instead of `models.ProviderType` function argument, I refactored it to simply accept `string` because in the end, we convert the provider type to string anyway. This removes the need for import. And the same applies for everything, well, except `version` which is just a simple package with constants (no metrics there).